### ISSUE TITLE
Bad performance on a project that uses failure crate

### ISSUE THREAD

**Issue by vasily-kirichenko** [intellij-rust/intellij-rust#2498]
Repro https://github.com/vasily-kirichenko/failure_perf_issue

Enter a space somewhere in `main.rs`. Result: CPU is very high load for long time:

![_1](https://user-images.githubusercontent.com/873919/39365265-4cc48240-4a39-11e8-8a3f-ce042ff24372.gif)

**Comment by Undin** [comment]
@vasily-kirichenko Is macro expansion enabled?

**Comment by vasily-kirichenko** [comment]
@Undin Yes. I've measured the time it needs to update syntax/error highlighting (cargo check is turned off):

* (Windows) no macro expansion - 7 seconds
* (Windows) macro expansion - 11 seconds
* (Mac) no macro expansion - 3 seconds
* (Mac) macro expansion - 4 seconds

Apart from the macro expansion influence (however, not that big), I'm curious why such a difference between windows and mac? The CPU are quite similar: i7-6700 (Windows) and i5-7600 (Mac).

This is how it works on the same code on Mac:

![untitled](https://user-images.githubusercontent.com/873919/39371268-d6e1e78c-4a49-11e8-94c7-e79285b572d6.gif)

**Comment by Undin** [comment]
@vasily-kirichenko We've just released a new plugin version with some performance improvements. Could you please check it and give us feedback?
